packages/dashboard/src/pages/Settings/api.ts

```diff
@@ -1,10 +1,23 @@
 import { createGlobalState } from '@masknet/shared'
 import { Services, Messages } from '../../API'
-
+import type { DataProvider } from '@masknet/public-api'
 export const [useLanguage] = createGlobalState(Services.Settings.getLanguage, (x) =>
     Messages.events.languageSettings.on(x),
 )
 
-export const [useTrendingDataSource] = createGlobalState(Services.Settings.getTrendingDataSource, (x) =>
+export const [useTrendingDataSource] = createGlobalState<DataProvider>(Services.Settings.getTrendingDataSource, (x) =>
     Messages.events.currentTrendingDataProviderSettings.on(x),
 )
+
+export const [useEthNetworkTradeProvider] = createGlobalState(Services.Settings.getEthNetworkTradeProvider, (x) =>
+    Messages.events.ethNetworkTradeProviderSettings.on(x),
+)
```

Comment:
why not
```suggestion
export const [useEthNetworkTradeProvider] = createGlobalState(
    Services.Settings.getEthNetworkTradeProvider,
    Messages.events.ethNetworkTradeProviderSettings.on,
)
```

Reply:
Oh, I remember. This function is used to prevent circular dependencies on initialization. I'll add document for that

Reply:
Hmm, Services are eagerly used. So I guess it is OK to early use Messages too.